test/unit/test_manifest_download.py

```diff
@@ -79,18 +88,43 @@ def test_manifest_download(self, download_func):
         self.assertEqual(download_func.call_count, len(self.manifest) - 1)
         self._assert_all_files_downloaded()
 
-    @patch('hca.dss.DSSClient._download_file', side_effect=_fake_download)
+    @patch('hca.dss.DSSClient._download_file', side_effect=_fake_download_file)
     def test_manifest_download_partial(self, _):
         _touch_file(self.dss._file_path(self.manifest[1][4]))
         self.dss.download_manifest_v2('manifest.tsv', 'aws')
         self._assert_all_files_downloaded()
 
     @patch('logging.Logger.warning')
     @patch('hca.dss.DSSClient._download_file', side_effect=[None, ValueError(), KeyError()])
-    def test_manifest_download_failed(self, download_func, warning_log):
+    def test_manifest_download_failed(self, _, warning_log):
         self.assertRaises(RuntimeError, self.dss.download_manifest_v2, 'manifest.tsv', 'aws')
         self.assertEqual(warning_log.call_count, 2)
 
+    @patch('hca.dss.DSSClient._get_file', side_effect=_fake_get_file)
+    def test_manifest_download_parallel(self, _):
```

Comment:
I think this is a useful test. The different patches make my head spin though. Can you achieve the same thing with a mock manifest that mentions the same file three times? You could put a time.sleep() somewhere to increase the probability that all the three threads are in the body of the atomic_write CM at the same time.

Reply:
I'll think about a way to make this more clear.